tests/scenario_tests/test_view_submission.py

```diff
@@ -15,6 +15,133 @@
+        "enterprise_id": "E111",
+        "enterprise_name": "Sandbox Org",
+    },
+    "user": {
+        "id": "W111",
+        "username": "primary-owner",
+        "name": "primary-owner",
+        "team_id": "T111",
+    },
+    "api_app_id": "A111",
+    "token": "verification_token",
+    "trigger_id": "111.222.valid",
+    "view": {
+        "id": "V111",
+        "team_id": "T111",
+        "type": "modal",
+        "blocks": [
+            {
+                "type": "input",
+                "block_id": "hspI",
+                "label": {
+                    "type": "plain_text",
+                    "text": "Label",
+                },
+                "optional": False,
+                "element": {"type": "plain_text_input", "action_id": "maBWU"},
+            }
+        ],
+        "private_metadata": "This is for you!",
+        "callback_id": "view-id",
+        "state": {
+            "values": {"hspI": {"maBWU": {"type": "plain_text_input", "value": "test"}}}
+        },
+        "hash": "1596530361.3wRYuk3R",
+        "title": {
+            "type": "plain_text",
+            "text": "My App",
+        },
+        "clear_on_close": False,
+        "notify_on_close": False,
+        "close": {
+            "type": "plain_text",
+            "text": "Cancel",
+        },
+        "submit": {
+            "type": "plain_text",
+            "text": "Submit",
+        },
+        "previous_view_id": None,
+        "root_view_id": "V111",
+        "app_id": "A111",
+        "external_id": "",
+        "app_installed_team_id": "T111",
+        "bot_id": "B111",
+    },
+    "response_urls": [],
+}
+
+raw_body = f"payload={quote(json.dumps(body))}"
+
+
+def simple_listener(ack, body, payload, view):
+    assert body["trigger_id"] == "111.222.valid"
+    assert body["view"] == payload
+    assert payload == view
+    assert view["private_metadata"] == "This is for you!"
+    ack()
+
+
+response_url_payload_body = {
+    "type": "view_submission",
+    "team": {"id": "T111", "domain": "test-test-test"},
+    "user": {
+        "id": "U111",
+        "username": "test-test-test",
+        "name": "test-test-test",
+        "team_id": "T111",
+    },
+    "api_app_id": "A111",
+    "token": "verification-token",
+    "trigger_id": "111.222.xxx",
+    "view": {
+        "id": "V111",
+        "team_id": "T111",
+        "type": "modal",
+        "blocks": [],
+        "callback_id": "view-id",
+        "state": {},
+        "title": {
+            "type": "plain_text",
+            "text": "My App",
+        },
+        "close": {
+            "type": "plain_text",
+            "text": "Cancel",
+        },
+        "submit": {
+            "type": "plain_text",
+            "text": "Submit",
+        },
+        "previous_view_id": None,
+        "root_view_id": "V111",
+        "app_id": "A111",
+        "external_id": "",
+        "app_installed_team_id": "T111",
+        "bot_id": "B111",
+    },
+    "response_urls": [
```

Comment:
This is the field this pull request takes care of. This array can have an element only when a modal has a conversations_select block element with response_url_enabled: true option.